skymap_scanner/utils/data_handling.py

```diff
@@ -55,24 +58,48 @@ def fetch_file(self, basename: str):
         Raises:
             RuntimeError: if the file retrieval fails.
         """
-        local_destination_path = self.staging_path / basename
-        http_source_path = f"{self.remote_path}/{basename}"
-        # not sure why we use the -O pattern here
-        cmd = [
-            "wget",
-            "-nv",
-            "-t",
-            "5",
-            "-O",
-            str(local_destination_path),
-            http_source_path,
-        ]
-
-        subprocess.run(cmd, check=True)
-
-        if not local_destination_path.is_file():
+        dest = self.staging_path / basename
+        url = f"{self.remote_path}/{basename}"
+
+        def backoff_sleep(attempt: int):
+            """Sleep with exponential backoff."""
+            sleep_duration = 2**attempt  # Exponential backoff: 2, 4, 8 seconds...
+            LOGGER.info(f"Retrying file download in {sleep_duration} seconds...")
+            time.sleep(sleep_duration)
+
+        # Step 1: Download the file
+        for attempt in itertools.count(1):
+            if attempt > 1:
+                backoff_sleep(attempt)
+            # get
+            try:
+                response = requests.get(
+                    url,
+                    stream=True,
+                    timeout=cfg.REMOTE_DATA_DOWNLOAD_TIMEOUT,
+                )
+                response.raise_for_status()  # Check if the request was successful (2xx)
+                break
+            except requests.exceptions.RequestException as e:
+                if attempt > cfg.REMOTE_DATA_DOWNLOAD_RETRIES:  # 'attempt' is 1-indexed
+                    raise RuntimeError(
+                        f"Download failed after {cfg.REMOTE_DATA_DOWNLOAD_RETRIES} retries: {e}"
+                    ) from e
+
+        # Step 2: Write the file
+        try:
+            with open(dest, "wb") as file:
+                for chunk in response.iter_content(chunk_size=8192):
```

Comment:
Anything special about 8192?

Reply:
Nothing special, it just limits how much is read into memory. I don't know how large our downloads are.